src/c4/C4_MPI.hh

```diff
@@ -85,7 +85,7 @@ void broadcast(ForwardIterator first, ForwardIterator last,
   Remember(int check =) broadcast(&size, 1, 0);
   Check(check == MPI_SUCCESS);
 
-  value_type *buf = new value_type[size];
+  auto *buf = new value_type[size];
   if (node() == 0)
```

Comment:
Hmmm, is replacing a type with `auto` one of the new best practices? I think this tends to obscure code.

Reply:
`auto` is only recommended in specific cases where the type is clearly identified on the RHS.  (1) when using `new`, and (2) when assigning via `static_cast`. Examples:

```c++
// Case 1
auto *buf = new int[size]`

// Case 2
auto bar = static_cast<double>(get_data());
```

Reply:
Got it.

Reply:
Interesting! That's a logical use case :)